src/main/java/org/mockito/Mockito.java

```diff
@@ -168,6 +169,52 @@
  * <p>
  * For more information about inline mock making, see <a href="#39">section 39</a>.
  *
+ * <h3 id="0.2">0.2. <a class="meaningful_link" href="#mockito-instrumentation" name="mockito-instrumentation">Explicitly setting up instrumentation for inline mocking (Java 21+)</a></h3>
+ *
+ * Starting from Java 21, the JDK restricts the ability of libraries to attach a Java agent to their own JVM.
+ * As a result, the inline-mock-maker might not be able to function without an explicit setup to enable
+ * instrumentation, and the JVM will always display a warning.
```

Comment:
**suggestion:** I propose to mention the JEP and use some of its wording, feel free to reword.

```suggestion
 * Starting from Java 21, the JDK prepares for a future release that restricts the 
 * ability of libraries to attach a Java agent to their own JVM (see <a href="https://openjdk.org/jeps/451">JEP 451</a>). 
 * As a result, the inline-mock-maker might not be able to function without an explicit setup to enable
 * instrumentation, and the JVM will always display a warning or emit an error in a future release.
```

Reply:
outdated